File: typescript/structural/flyweight/guru_example/flyweight.ts
export class Flyweight {
  private readonly sharedState: any

  constructor(sharedState: any) {
    this.sharedState = sharedState
  }

  public operation(uniqueState): void {
    const s = JSON.stringify(this.sharedState)
    const u = JSON.stringify(uniqueState)

    console.log(`Flyweight: \n  shared state = ${s}, \n  unique state = ${u}`)
  }
}

File: dart/structural/flyweight/guru_example/bin/flyweight.dart
import 'dart:convert';

class Flyweight {
  dynamic _sharedState;

  Flyweight(dynamic ss) {
    _sharedState = ss;
  }

  void operation(dynamic uniqueState) {
    final s = jsonEncode(_sharedState);
    final u = jsonEncode(uniqueState);
    print('Flyweight:\n  shared state = $s\n  unique state = $u');
  }
}
